### PR TITLE
Default aliasing bug fix

### LINKED ISSUE #4827: When aliasing default and targeting ES3, TS does not quote the object accessor
Given `foosert.ts`:
```ts
export class Foo {
	static CONSTANT = "Foo";
}

export default function assert(value: boolean) {
	if (!value) throw new Error("Assertion failed!");
}
```
and `entrypoint.ts`:
```ts
import {Foo, default as assert} from "./foosert";
assert(Foo.CONSTANT === "Foo");
```
When running `tsc -m 'commonjs' entrypoint.ts`, `entrypoint.ts` gets compiled to:
```js
var foosert_1 = require("./foosert");
foosert_1.default(foosert_1.Foo.CONSTANT === "Foo");
```

but should be:
```js
var foosert_1 = require("./foosert");
foosert_1["default"](foosert_1.Foo.CONSTANT === "Foo");
```

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #4827

### CODE REVIEW

**Comment by weswigham** [comment]
@DanielRosenwasser you wanna check this over before merging it?

**Comment by mhegazy** [comment]
:+1:

File: src/compiler/emitter.ts [hunk 1]
```diff
@@ -1515,8 +1515,14 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                         else if (declaration.kind === SyntaxKind.ImportSpecifier) {
                             // Identifier references named import
                             write(getGeneratedNameForNode(<ImportDeclaration>declaration.parent.parent.parent));
-                            write(".");
-                            writeTextOfNode(currentSourceFile, (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name);
+                            var name =  (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name;
+                            var identifier = getSourceTextOfNodeFromSourceFile(currentSourceFile, name);
+                            if (languageVersion == ScriptTarget.ES3 && identifier === "default") {
+                                write(`["default"]`);
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: Just use single quotes.

File: src/compiler/emitter.ts [hunk 2]
```diff
@@ -1515,8 +1515,14 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                         else if (declaration.kind === SyntaxKind.ImportSpecifier) {
                             // Identifier references named import
                             write(getGeneratedNameForNode(<ImportDeclaration>declaration.parent.parent.parent));
-                            write(".");
-                            writeTextOfNode(currentSourceFile, (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name);
+                            var name =  (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name;
+                            var identifier = getSourceTextOfNodeFromSourceFile(currentSourceFile, name);
+                            if (languageVersion == ScriptTarget.ES3 && identifier === "default") {
+                                write(`["default"]`);
+                            } else {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: `else` on the next lien

File: src/compiler/emitter.ts [hunk 3]
```diff
@@ -1515,8 +1515,14 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
                         else if (declaration.kind === SyntaxKind.ImportSpecifier) {
                             // Identifier references named import
                             write(getGeneratedNameForNode(<ImportDeclaration>declaration.parent.parent.parent));
-                            write(".");
-                            writeTextOfNode(currentSourceFile, (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name);
+                            var name =  (<ImportSpecifier>declaration).propertyName || (<ImportSpecifier>declaration).name;
+                            var identifier = getSourceTextOfNodeFromSourceFile(currentSourceFile, name);
+                            if (languageVersion == ScriptTarget.ES3 && identifier === "default") {
```
Comment (DanielRosenwasser) [inline src/compiler/emitter.ts]: `===`

File: src/compiler/emitter.ts (hunk 1 above)
Comment (weswigham) [inline src/compiler/emitter.ts]: But that would fail the linter.

File: src/compiler/emitter.ts (hunk 2 above)
Comment (weswigham) [inline src/compiler/emitter.ts]: :+1:

File: src/compiler/emitter.ts (hunk 3 above)
Comment (weswigham) [inline src/compiler/emitter.ts]: :+1:

File: src/compiler/emitter.ts (hunk 1 above)
Comment (weswigham) [inline src/compiler/emitter.ts]: On that note, we should consider merging #4458 at some point.

**Comment by vladima** [comment]
:+1:

**Pull request merged**